core/src/test/java/uk/gov/gchq/koryphe/impl/function/IterableFunctionTest.java

```diff
@@ -92,4 +94,33 @@ public void shouldReturnEmptyIterableForEmptyInput() {
         assertNotNull(result);
         assertTrue(Iterables.isEmpty(result));
     }
+
+    @Test
+    public void shouldPopulateFunctionFromBuilder() {
+        // Given
+        final IterableFunction<Integer, String> function = new IterableFunction.Builder<Integer>()
+                .first(Object::toString)
+                .then(Integer::valueOf)
+                .then(Object::toString)
+                .build();
+
+        // Then
+        assertEquals(3, function.getFunctions().size());
+    }
+
+    @Test
+    public void shouldApplyMultipleFunctions() {
```

Comment:
Please can you add some test for some of the other edge cases, like null functions, functions where the input/output types are incorrect, empty functions.